cli/diagnostics.rs

```diff
@@ -0,0 +1,754 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+//! This mod adds source maps and ANSI color display to deno::DenoDiagnostic.
+use crate::ansi;
+use deno;
+use deno::DenoDiagnostic;
+use deno::DenoDiagnosticCategory;
+use deno::DenoDiagnosticFrame;
+use deno::DenoDiagnosticSources;
+use source_map_mappings::parse_mappings;
+use source_map_mappings::Bias;
+use source_map_mappings::Mappings;
+use std::collections::HashMap;
+use std::fmt;
+use std::str;
+
+/// Wrapper around DenoDiagnostic which provides color to_string.
+pub struct DenoDiagnosticColor<'a>(pub &'a DenoDiagnostic);
+
+struct DenoDiagnosticFrameColor<'a>(&'a DenoDiagnosticFrame);
+
+pub trait SourceMapGetter {
+  /// Returns the raw source map file.
+  fn get_source_map(&self, script_name: &str) -> Option<Vec<u8>>;
+}
+
+/// Cached filename lookups. The key can be None if a previous lookup failed to
+/// find a SourceMap.
+type CachedMaps = HashMap<String, Option<SourceMap>>;
+
+struct SourceMap {
+  mappings: Mappings,
+  sources: Vec<String>,
+}
+
+impl<'a> fmt::Display for DenoDiagnosticFrameColor<'a> {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    let frame = self.0;
+    // Note when we print to string, we change from 0-indexed to 1-indexed.
```

Comment:
Can you move this comment to `format_script_line_column` and add short explanation why indexing changes?